api/v1alpha1/api.go

```diff
@@ -47,18 +47,68 @@ type LLMRouteSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self.schema == 'OpenAI'"
 	APISchema LLMAPISchema `json:"inputSchema"`
-	// HTTPRoute is the base HTTPRouteSpec (https://gateway-api.sigs.k8s.io/api-types/httproute/) in
-	// the Gateway API on which this LLMRoute will be implemented. AI Gateway controller will generate a HTTPRoute based
-	// on the configuration given here with the additional modifications to achieve the necessary jobs,
-	// notably inserting the AI Gateway external processor filter.
+	// Rules is the list of LLMRouteRule that this LLMRoute will match the traffic to.
+	// Each rule is a subset of the HTTPRoute in the Gateway API (https://gateway-api.sigs.k8s.io/api-types/httproute/).
 	//
-	// In the matching rules in the HTTPRoute here, `x-envoy-ai-gateway-llm-model` header
-	// can be used to describe the routing behavior.
+	// AI Gateway controller will generate a HTTPRoute based on the configuration given here with the additional
```

Comment:
How are we going to generate the HTTPRoute as certain feature like traffic split can't be used on the underlying HTTPRoute ? I guess we can only translate a subset of the matching rules from `LLMRouteRules` to `HTTPRouteRules` which does not include weights etc.

Reply:
i am thinking we will perform traffic splitting at the extproc level, and chose the backend

Reply:
If I understand correctly, the ext proc resolves the llm backend using the matching rules and the HttpRoute resource should be just created as simple as routing to the backends with the backend http header. Is that correct ?

Reply:
exactly, that's what i am thinking to implement this thingy